source/content/pantheon-yml.md

```diff
@@ -148,6 +147,12 @@ This table shows the recommended MariaDB version for each CMS:
 
 Users of Drupal 6 sites should consider [upgrading to Drupal 7](/drupal-updates#upgrade-from-drupal-6-to-drupal-7) for better support.
 
+#### Considerations
+
+The default database version for new sites is MariaDB version 10.4.  If your site has any older contrib modules that have issue with MariaDB 10.4, you may set the MariaDB version to 10.3 in your pantheon.yml file. This should be rare.
+
+Note that Drupal 9 **requires** MariaDB 10.4. If you have a Drupal 8 site that you are planning on upgrading to Drupal 9, ensure that the database has been upgraded to 10.4 in all environments before beginning the Drupal 9 upgrade.
```

Comment:
I'm wondering about this language maybe even being a bit stronger still?  "has been _fully_ upgraded"?  Is it going too far to add "The MariaDB upgrade is not instant, so applying both upgrades simultaneously could result in temporary downtime"?

Reply:
Yeah, and we could get even longer if we talked about the Drupal bug that sometimes mis-reports the PHP version. Not sure how in-depth we should be here.

Reply:
@greg-1-anderson Correct me if I am wrong but Drupal9 can run on 10.3 as well. We just strongly prefer upgrading to 10.4

Reply:
That is incorrect. In the early days, before 10.4 is available, we patched Drupal 9 to remove the check. Out of the box, Drupal 9.0.0 and later refuse to run unless MariaDB is 10.4 or higher. Contrib modules may rely on this requirement, regardless of Drupal core version.

Reply:
@greg-1-anderson is it a Pantheon limitation? I ask because product is under the impression that 10.3 is the requirement as posted here https://www.drupal.org/docs/system-requirements/database-server-requirements#s-mysql-mariadb-or-percona-server-recommended

Reply:
You're right! My bad memory. Drupal 9 requires 10.3+. It's just 10.0 that isn't kosher.

Really, we could default all of these upstreams to 10.3. That would be a lot safer. Well, we already default 9 to 10.4, so I guess there's no point in rolling that one back.